packages/common/core-interfaces/src/test/jsonString.spec.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	jsonStringOfString,
+	jsonStringOfObjectWithArrayOfNumbers,
+	jsonStringOfStringRecordOfNumbers,
+	jsonStringOfStringRecordOfNumberOrUndefined,
+	jsonStringOfBigInt,
+	jsonStringOfUnknown,
+} from "./testValues.js";
+
+import type { JsonString } from "@fluidframework/core-interfaces/internal";
+import { JsonStringify } from "@fluidframework/core-interfaces/internal";
+
+function parameterAcceptedAs<T>(_t: T): void {
+	// Do nothing.  Used to verify type compatibility.
+}
+
+const jsonStringOfLiteral = JsonStringify("literal");
+
+describe("JsonString", () => {
+	it("`JsonString<string>` is assignable to `string`", () => {
+		parameterAcceptedAs<string>(jsonStringOfString);
+		parameterAcceptedAs<string>(jsonStringOfLiteral);
+		parameterAcceptedAs<string>(jsonStringOfObjectWithArrayOfNumbers);
+		parameterAcceptedAs<string>(jsonStringOfStringRecordOfNumbers);
+		parameterAcceptedAs<string>(jsonStringOfStringRecordOfNumberOrUndefined);
+		parameterAcceptedAs<string>(jsonStringOfBigInt);
+		parameterAcceptedAs<string>(jsonStringOfUnknown);
+	});
+
+	it("`string` is not assignable to `JsonString<unknown>`", () => {
+		parameterAcceptedAs<JsonString<unknown>>(
+			// @ts-expect-error Type 'string' is not assignable to type 'JsonString<unknown>'
+			"a string",
+		);
+	});
+
+	it("object is not assignable to `JsonString<unknown>`", () => {
+		parameterAcceptedAs<JsonString<unknown>>(
+			// @ts-expect-error Type '{ property: string; }' is not assignable to type 'JsonString<unknown>'
+			{ property: "value" },
+		);
+	});
+
+	describe("is variant by T", () => {
```

Comment:
Just reminder about Craig's comment during Hackathon demo meeting, in case the variance descriptions haven't been updated.